### PR TITLE
Add Visibility field to ServiceSpec

### PR DESCRIPTION
**Pull request by justinsb**
Visibility = Cluster replaces CreateExternalLoadBalancer = false
Visibility = LoadBalacner replaces CreateExternalLoadBalancer = true

(Visibility = NodePort is coming, and maybe more in future!)

### CODE REVIEW

**Comment by justinsb** [comment]
@thockin for your consideration :-)

File: pkg/kubectl/cmd/expose.go [hunk 1]
```diff
@@ -61,7 +61,8 @@ func NewCmdExposeService(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().String("protocol", "TCP", "The network protocol for the service to be created. Default is 'tcp'.")
 	cmd.Flags().Int("port", -1, "The port that the service should serve on. Required.")
 	cmd.MarkFlagRequired("port")
-	cmd.Flags().Bool("create-external-load-balancer", false, "If true, create an external load balancer for this service. Implementation is cloud provider dependent. Default is 'false'.")
```
Comment (thockin) [inline pkg/kubectl/cmd/expose.go]: I think you have to keep the old flag for a while.

**Comment by thockin** [comment]
LGTM but compat flag.  I assume you're running each piece of this transform through manual testing (especially where flags and outputs and such things are concerned)?

**Comment by thockin** [comment]
Failed CI: 

# github.com/GoogleCloudPlatform/kubernetes/pkg/cloudprovider/servicecontroller
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/cloudprovider/servicecontroller/servicecontroller_test.go:190: cannot use true (type bool) as type api.ServiceVisibility in argument to newService
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/cloudprovider/servicecontroller [build failed]

**Comment by vmarmol** [comment]
You'll also need a rebase it seems.

File: pkg/kubectl/cmd/expose.go (hunk 1 above)
Comment (justinsb) [inline pkg/kubectl/cmd/expose.go]: I did (I think you did, actually); it just moved down a line!  And I think we continue to pass create-external-load-balancer to the helper, which converts it to a visibility flag.

**Comment by justinsb** [comment]
Rebased & renamed per suggestions.

Have been running unit tests (and there are a few additional unit tests that I have had to remove because they rely on multiple PRs) & have been manual testing the final product.  I think on this PR I was trying to save time by having Travis check it for me, but then I forgot to circle back and check that it passed.

**Comment by thockin** [comment]
LGTM.

Do not merge until all load-balancer PRs are sequenced.

**Comment by justinsb** [comment]
NodePort is now merged, in a parallel PR

**Pull request closed without merging**